aspnetcore/mvc/razor-pages/index.md

```diff
@@ -69,398 +67,254 @@ The associations of URL paths to pages are determined by the page's location in
+attribute to generate a link to the Edit page. The link contains route data with the contact ID. For example, `http://localhost:5000/Edit/1`.
 
-*MyApp/Pages/Contact.cshtml.cs*
+The *Pages/Edit.cshtml* file:
 
-```c#
-using Microsoft.AspNetCore.Mvc;
-using Microsoft.AspNetCore.Mvc.RazorPages;
+[!code-html[main](index/sample/RazorPagesContacts/Pages/Edit.cshtml?highlight=1)]
 
-namespace MyApp.Pages
-{
-    public class ContactModel : PageModel
-    {
-        public ContactModel(ApplicationDbContext db)
-        {
-            Db = db;
-        }
-
-        [BindProperty]
-        public Contact Contact { get; set; }
-        
-        private ApplicationDbContext Db { get; }
-    
-        public async Task<IActionResult> OnPostAsync()
-        {
-            if (ModelState.IsValid)
-            {
-                Db.Contacts.Add(Contact);
-                await Db.SaveChangesAsync();
-                return RedirectToPage();
-            }
-            
-            return Page();
-        }
-    }
-}
-```
+<!-- REVIEW -->
+The first line contains the `@page "{id:int}"` directive. `"{id:int}"` tells the page to accept requests to the page that contain `int` route data. If a request to the page doesn't contain route data that can be converted to an `int`, the runtime returns an HTTP 404 (not found) error.
+
+The *Pages/Edit.cshtml.cs* file:
+
+[!code-cs[main](index/sample/RazorPagesContacts/Pages/Edit.cshtml.cs)]
 
-By convention, the `PageModel` class is called `<PageName>Model` and is in the same namespace as the page. Not much change is needed to convert from a page using `@functions` to define handlers and a page using a `PageModel` class. The main change is to add constructor injection for all your injected (`@inject`) properties.
+<a name="xsrf"></a>
 
-Using a `PageModel` supports unit testing, but requires you to write an explicit constructor and class. Pages without `PageModel` files support runtime compilation, which can be an advantage in development.
+## XSRF/CSRF and Razor Pages
 
-## Using the view engine
+You don't have to write any code for [antiforgery validation](xref:security/anti-request-forgery). Antiforgery token generation and validation is automatically included in Razor Pages.
 
-Pages work with all the features of the Razor view engine. Layouts, partials, templates, tag helpers, *_ViewStart.cshtml*, *_ViewImports.cshtml* all work in the same way they do for conventional Razor views. 
+## Using Layouts, partials, templates, and tag helpers with Razor Pages
+
+Pages work with all the features of the Razor view engine. Layouts, partials, templates, tag helpers, *_ViewStart.cshtml*, *_ViewImports.cshtml* work in the same way they do for conventional Razor views. 
 
 Let's declutter this page by taking advantage of some of those features. 
 
-Add a layout page for the HTML skeleton, and set the `Layout` property from `_ViewStart.cshtml`:
+Add a [layout page](xref:mvc/views/layout) to Pages/_Layout.cshtm:
```

Comment:
~~Add a [layout page](xref:mvc/views/layout) to Pages/_Layout.cshtm:~~
Add a [layout page](xref:mvc/views/layout) to *Pages/_Layout.cshtml*: